src/query/functions-v2/src/scalars/comparison.rs

```diff
@@ -39,89 +39,259 @@ use regex::bytes::Regex;
 use crate::scalars::string_multi_args::regexp;
 
 pub fn register(registry: &mut FunctionRegistry) {
-    register_simple_cmp::<StringType>(registry);
-    register_simple_cmp::<TimestampType>(registry);
+    register_string_cmp(registry);
+    register_date_cmp(registry);
     register_boolean_cmp(registry);
     register_number_cmp(registry);
     register_variant_cmp(registry);
     register_like(registry);
 }
 
-fn register_simple_cmp<T: ArgType>(registry: &mut FunctionRegistry)
-where for<'a> T::ScalarRef<'a>: PartialOrd + PartialEq {
-    registry.register_2_arg::<T, T, BooleanType, _, _>(
+#[inline(always)]
+fn all_true_domain() -> BooleanDomain {
+    BooleanDomain {
+        has_true: true,
+        has_false: false,
+    }
+}
+
+#[inline(always)]
+fn all_false_domain() -> BooleanDomain {
+    BooleanDomain {
+        has_true: false,
+        has_false: true,
+    }
+}
+
+fn register_string_cmp(registry: &mut FunctionRegistry) {
+    registry.register_2_arg::<StringType, StringType, BooleanType, _, _>(
         "eq",
         FunctionProperty::default(),
         |_, _| FunctionDomain::NoThrow,
```

Comment:
@andylokandy  Is it recommended to use `FunctionDomain::Domain(BooleanDomain {
                has_false: true,
                has_true: true,
            })`  or  `FunctionDomain::NoThrow` ?

Reply:
`FunctionDomain::NoThrow` is equivalent to `FunctionDomain::Domain(T::full_domain())`.

Reply:
Yes, let's do this in another pr. Currently it's ok in this one.